src/kafka/kafka_producer.py

```diff
@@ -0,0 +1,136 @@
+                    {
+                        "Location" - has not added yet
+                        "time" - TimeStamp
+                        "barometric reading"
+                        "gyroscope reading x" - rotation around x-axis
+                        "gyroscope reading y" - rotation around y-axis
+                        "Wind speed"
+                        }
+    """
+
+    def __init__(self, address, n):
+        """
+            Intialize variables
+            ndrones = Number of drones
+            event_log = Logical which initiates an anomalous events
+            event_log_time = time parameter for anamalous events
+            __barometer_event_reading = barometric event resulting from anomalous event
+        """
+        self.dataProducer = KafkaProducer(bootstrap_servers=address)
+        self.ndrones = n
+        self.event_log = np.zeros(n, dtype = bool)
+        self.event_log_time = -5*np.ones(n)
+        self.__barometer_event_reading = np.zeros(n)
+
+    def Serialize_JSON(self):
+        """
+            Serialize datetime JSON for kafka producer
+        """
+        if isinstance(self.TimeStamp, datetime):
+            return self.TimeStamp.__str__()
+
+    def instantiate_event(self):
+        """
+            instantiate anomalous event
+        """
+        for i in range(self.ndrones):
+            if np.random.uniform(0,1) < 0.01:
+                self.event_log[i] = True
+
+    def stop_event(self):
+        """
+            Stop anamalous event
+        """
+        for i in range(self.ndrones):
+            if self.event_log[i] and self.event_log_time[i]>=5:
+                self.event_log[i] = False
+                self.event_log_time[i] = -5
+
+    def update_time_log(self):
+        dt = 0.1
+        for i in range(self.ndrones):
+            if self.event_log[i]:
+                self.event_log_time += dt
+
+    def generate_event(self):
+        """
+            simulate __barometer_event_reading from anamalous event
+        """
+        self.instantiate_event()
+        self.update_time_log()
+        for i in range(self.ndrones):
+            if self.event_log[i]:
+                ts = self.event_log_time[i]
+                self.__barometer_event_reading[i] = recovery_event(1, 1, 100, ts, 1)\
+                                                    .generate_altitude()
+
+    def ProduceData(self, topic):
+        """
+            Produce data and sent to kafka producer
+        """
+        self.TimeStamp = datetime.now()
+        print(self.TimeStamp)
+        while True:
+            self.generate_event()
+            baromatric_reading = np.random.uniform(395, 405, self.ndrones) + self.__barometer_event_reading
+            latitude = np.zeros(self.ndrones)
+            longitude = np.zeros(self.ndrones)
+            gyrometer_x = np.random.uniform(-0.4, 0.4, self.ndrones)
+            gyrometer_y = np.random.uniform(-0.4, 0.4, self.ndrones)
+            wind_speed = 55*np.ones(self.ndrones)
```

Comment:
These are either random from uniform or all uniform--need to change to gaussian distributions at least.